### PR TITLE
Water uptake final.

### PR DESCRIPTION
**Pull request by pressel**
This should be the last PR for water uptake! This PR brings a port of `modal_aero_water_uptake_dr`, which is a top level driver.

### CODE REVIEW

File: src/mam4xx/water_uptake.hpp [hunk 1]
```diff
@@ -429,7 +429,102 @@ void modal_aero_wateruptake_dryaer(
...
+  // compute aerosol wet radius, volume, diameter and aerosol water
+  Real wetrad[AeroConfig::num_modes()];
+  Real qaerwat[AeroConfig::num_modes()];
+  modal_aero_water_uptake_wetaer(rhcrystal, rhdeliques, dgncur_a, dryrad, hygro,
+                                 rh, naer, dryvol, wetrad, wetvol, wtrvol,
+                                 dgncur_awet, qaerwat);
+}
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_water_uptake_dr(
```
Comment (odiazib) [inline src/mam4xx/water_uptake.hpp:476]: Does this function corresponds to the subroutine `modal_aero_wateruptake_dr` from [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/67ae6cd6f8b8381c3162b9071fe39754ab95fce7/components/eam/src/chemistry/utils/modal_aero_wateruptake.F90#L127).

**Review by odiazib — APPROVED** [review]

File: src/mam4xx/water_uptake.hpp [hunk 2]
```diff
@@ -429,7 +429,102 @@ void modal_aero_wateruptake_dryaer(
   }
 }
 
+KOKKOS_INLINE_FUNCTION
+void modal_aero_water_uptake_dr_b4_wetdnes(
```
Comment (jeff-cohere) [inline src/mam4xx/water_uptake.hpp:433]: Should this be `wetdens` instead of `wetdnes`?

File: src/mam4xx/water_uptake.hpp [hunk 3]
```diff
@@ -429,7 +429,102 @@ void modal_aero_wateruptake_dryaer(
...
+    Real dgncur_a[AeroConfig::num_modes()],
+    Real dgncur_awet[AeroConfig::num_modes()],
+    Real wetdens[AeroConfig::num_modes()]) {
+
+  Real drymass[AeroConfig::num_modes()];
+  Real specdens_1[AeroConfig::num_modes()];
+  Real wetvol[AeroConfig::num_modes()];
+  Real wtrvol[AeroConfig::num_modes()];
+
+  modal_aero_water_uptake_dr_b4_wetdnes(nspec_amode, specdens_amode, spechygro,
```
Comment (jeff-cohere) [inline src/mam4xx/water_uptake.hpp:490]: `wetdens`?

File: src/mam4xx/water_uptake.hpp [hunk 4]
```diff
@@ -429,7 +429,102 @@ void modal_aero_wateruptake_dryaer(
...
+    Real state_q[nvars], Real temperature, Real pmid, Real cldn,
+    Real dgncur_a[AeroConfig::num_modes()],
+    Real dgncur_awet[AeroConfig::num_modes()]) {
+
+  Real drymass[AeroConfig::num_modes()];
+  Real specdens_1[AeroConfig::num_modes()];
+  Real wetvol[AeroConfig::num_modes()];
+  Real wtrvol[AeroConfig::num_modes()];
+
+  modal_aero_water_uptake_dr_b4_wetdnes(nspec_amode, specdens_amode, spechygro,
```
Comment (jeff-cohere) [inline src/mam4xx/water_uptake.hpp:513]: `wetdens`?

**Review by jeff-cohere — APPROVED** [review]
Looks good. I found what looks like a typo.

File: src/mam4xx/water_uptake.hpp (hunk 1 above)
Comment (pressel) [reply src/mam4xx/water_uptake.hpp:476]: It does. I've added comments to make this clear.

**Pull request merged**